src/server/scripts/Northrend/VioletHold/violet_hold.cpp

```diff
@@ -675,8 +752,39 @@ struct violet_hold_trashAI : public npc_escortAI
     {
         instance = creature->GetInstanceScript();
         bHasGotMovingPoints = false;
-        portalLocationID = instance->GetData(DATA_PORTAL_LOCATION);
-        secondPortalRouteID = 0;
+
+
+        if (instance->GetData(DATA_MAIN_EVENT_PHASE) == NOT_STARTED)
+        {
+            if (Creature* portal = me->FindNearestCreature(NPC_TELEPORTATION_PORTAL, 10.0f))
+            {
+                uint64 portalGUID = portal->GetGUID();
+                for (uint8 i = 0; i < 3; i++)
```

Comment:
please use a define in the array declaration if you plan to iterate it in a for loop and use that define in the for loop too as upper bound